src/perf/best-practices.md

```diff
@@ -0,0 +1,375 @@
+the layout process can be expensive.
+For example, consider a 10x500 grid of `Card`s.
+A grid should have uniformly sized cells,
+so the layout code performs a pass,
+starting from the root of the grid (in the widget tree),
+asking each card in the visible grid to return
+its _intrinsic_ size&mdash;the size
+that the widget prefers, assuming no constraints.
+With this information,
+the framework calculates a uniform cell size,
+and re-visits the grid objects,
+_telling_ each card what size to use. 
+
+If your code performs unnecessary layout passes,
+it can slow performance. To determine if this
+is your issue, enable the **Track layouts option**
+in DevTools (disabled by default),
+and look at the app's [stack trace][]
+to learn how many layout passes were performed.
+[PENDING: Where is the track layout option in DevTools?]
+
+[PENDING: How does once determine "excessive"?]
+[PENDING: Is the primary way to fix this by lazy loading
+the grid or list? Or is there something else?]
+
+To dive even deeper into how layout works,
+check out the [layout and rendering][]
+section in the [Flutter architectural overview][].
+
+
+[Flutter architectural overview]: {{site.url}}/resources/architectural-overview
+[how layout and constraints work]: {{site.url}}/development/ui/layout/constraints
+[layout and rendering]: {{site.url}}/resources/architectural-overview#layout-and-rendering
+[stack trace]: {{site.url}}/development/tools/devtools/cpu-profiler#flame-chart
+
+---
+
+##  Build and display frames in 16ms
+
+Since there are two separate threads for building
+and rendering, you have 16ms for building,
+and 16ms for rendering on a 60Hz display.
+If latency is a concern,
+build and display a frame in 16ms _or less_.
+Note that means built in 8ms or less,
+and rendered in 8ms or less,
+for a total of 16ms or less.
+If missing frames (jankiness) is a concern,
+then 16ms for each of
+the build and render stages is OK.
+
+If your frames are rendering in well under
+16ms total in [profile mode][],
+you likely don’t have to worry about performance
+even if some performance pitfalls apply,
+but you should still aim to build and
+render a frame as fast as possible. Why?
+
+* Lowering the frame render time below 16ms might not make a visual
+  difference, but it **improves battery life** and thermal issues.
+* It might run fine on your device, but consider performance for the
+  lowest device you are targeting.
+* As 120fps devices become more widely available,
+  you’ll want to render frames in under 8ms (total)
+  in order to provide the smoothest experience.
+
+If you are wondering why 60fps leads to a smooth visual experience,
+check out the video [Why 60fps?][]
+
+[profile mode]: {{site.url}}/testing/build-modes#profile
+[Why 60fps?]: {{site.youtube-site}}/watch?v=CaMTIgxCSqU
+
+# Pitfalls
+
+If you need to tune your app’s performance,
+or perhaps the UI isn't as smooth as you expect,
+the Flutter plugin for your IDE can help.
```

Comment:
@kenzieschmoll Is this also available in devtools?

Reply:
I've added this, but was unsure if I should delete the IDE info. Please check, @kenzieschmoll